api/src/routes/user.ts

```diff
@@ -301,5 +303,59 @@ export const userRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/user/report-user',
+    {
+      schema: schemas.reportUser,
+      preValidation: (req, reply, done) => {
+        // NOTE: Is this the right hook?
+        req.body = {
+          ...req.body,
+          reportDescription: trimTags(req.body.reportDescription)
+        };
+        done();
+      }
+    },
+    async (req, reply) => {
+      try {
+        const user = await fastify.prisma.user.findUniqueOrThrow({
+          where: { id: req.session.user.id }
+        });
+        const { username, reportDescription: report } = req.body;
+
+        if (!username || !report || report === '') {
+          // NOTE: Do we want to log these instances?
+          void reply.code(400);
+          return {
+            type: 'danger',
+            message: 'flash.provide-username'
+          } as const;
+        }
```

Comment:
I'd handle this in validation, and just provide a custom errorHandler to create the desired response.
```suggestion
```